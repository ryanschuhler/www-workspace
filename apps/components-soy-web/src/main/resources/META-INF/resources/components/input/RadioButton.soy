{namespace RadioButton}

/**
 * RadioButton component
 */
{template .render}
  {@param id: string}
  {@param name: string}
  {@param? checked: bool}
  {@param? disabled: bool}
  {@param? label: string}
  {@param? onChange: any}
  {@param? onClick: any}
  {@param? required: bool}

  {let $inputAttributes kind="attributes"}
    {if $checked}
      checked="checked"
    {/if}

    {if $disabled}
      {sp}disabled{sp}
    {/if}

    class="mdc-radio__native-control"
    id="{$id}"
    name="{$name}"

    onClick="{$onClick}"
    onchange="{$onChange}"

    {if $required}
      required="true"
    {/if}

    type="radio"
  {/let}

  {let $radioClasses kind="text"}
    mdc-radio

    {if $checked}
      {sp}checked
    {/if}

    {if $disabled}
      {sp}mdc-radio--disabled
    {/if}
  {/let}

  <div class="mdc-form-field">
    <div class="{$radioClasses}">
      <input {$inputAttributes}/>

      <div class="mdc-radio__background">
        <div class="mdc-radio__outer-circle"></div>
        <div class="mdc-radio__inner-circle"></div>
      </div>
    </div>

    {if $label}
      <label for="{$id}" id="{$id}-label" class="mdc-radio-label">{$label}</label>
    {/if}
  </div>
{/template}


