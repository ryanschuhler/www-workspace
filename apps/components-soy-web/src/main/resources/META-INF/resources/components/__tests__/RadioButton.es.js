'use strict';

import RadioButton from '../input/RadioButton.es';

let component;

describe('RadioButton', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  it('should render', () => {
    component = new RadioButton({
      id: 'foo',
      name: 'bar'
    });

    expect(component).toMatchSnapshot();
  });
});