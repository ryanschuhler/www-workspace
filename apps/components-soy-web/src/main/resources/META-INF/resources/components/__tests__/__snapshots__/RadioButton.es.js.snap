// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render 1`] = `
<div class="mdc-form-field">
  <div class="mdc-radio">
    <input class="mdc-radio__native-control" id="foo" name="bar" type="radio">
    <div class="mdc-radio__background">
      <div class="mdc-radio__outer-circle"></div>
      <div class="mdc-radio__inner-circle"></div>
    </div>
  </div>
</div>
`;
